File: 1-two-sum/two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int,int>hash;
        for(int i=0;i<nums.size();i++){
            int val=target-nums[i];
            if(hash.find(val)!=hash.end()){
                return {hash[val],i};
            }
            hash[nums[i]]=i;
        }
        return {};
    }
};

File: 1-two-sum/two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> mp = new HashMap<>();
        int n = nums.length;
        int ans[] = new int[2];
        for(int i=0; i<n; i++){
            if(mp.containsKey(target - nums[i])){
                ans[0] = mp.get(target-nums[i]);
                ans[1] = i;
                return ans;
            }
            mp.put(nums[i], i);
        }
        return ans;
    }
}